tinygrad/codegen/kernel.py

```diff
@@ -8,6 +8,23 @@
 from tinygrad.shape.symbolic import sint
 from tinygrad.shape.view import strides_for_shape
 
+class TensorCore(NamedTuple):
```

Comment:
Consider `@dataclass(frozen=True)`. Newer, slightly faster.

Reply:
I don't ever want this. Stop suggesting it, it's harder to reason about in exchange for a stupidly small amount of speed. That tradeoff is never worth it.

Reply:
Thanks for the nuance, won't mention it again. I could have been more clear: I suggested a dataclass because it because it is newer, a bit more flexible and has more intuitive equalization (NamedTuple does not check the name/type!).

That last thing can lead to insidious bugs on NamedTuples with the same number of members, e.g.:

```
from typing import NamedTuple

def LazyOp(NamedTuple):
  r: int
  f: Tuple

def TensorCore(NamedTuple):
  g: int
  h: Tuple

print(LazyOp(1, (1,2,3)) == TensorCore(1,(1,2,3)))

```

https://stackoverflow.com/questions/51671699/data-classes-vs-typing-namedtuple-primary-use-case

Reply:
Changed in the latest commit